napalm_logs/server.py

```diff
@@ -28,36 +29,78 @@ def __init__(self,
         self.__pipe = pipe
         self.__os_pipe_map = os_pipe_map
         self.__up = False
+        self.compiled_prefixes = None
+
+    def _compile_prefixes(self):
+        '''
+        Create a dict of all OS prefixes and their compiled regexs
+        '''
+        self.compiled_prefixes = {}
+        for os, os_config in self.config.iteritems():
+            if not os_config:
+                continue
+            values = os_config.get('prefix', {}).get('values', {})
+            line = os_config.get('prefix', {}).get('line', '')
+            # We will now figure out which position each value is in so we can use it with the match statement
+            position = {}
+            for key in values.keys():
+                position[line.find('{' + key + '}')] = key
+            sorted_position = {}
+            for i, elem in enumerate(sorted(position.items())):
+                sorted_position[elem[1]] = i + 1
+
+            escaped = re.escape(line).replace('\{', '{').replace('\}', '}')
+            self.compiled_prefixes[os] = { 
+                'prefix': re.compile(escaped.format(**values)),
+                'prefix_positions': sorted_position,
+                'values': values,
+                'messages': {}
+                }
 
     def _identify_os(self, msg):
         '''
         Using the prefix of the syslog message,
         we are able to identify the operating system and then continue parsing.
         '''
-        return ('junos', 'crap')
+        ret = {}
+        for os, data in self.compiled_prefixes.iteritems():
```

Comment:
I would suggest to avoid the `os` var name, as `os` can be the name of the imported package.